lib/os/dec.c

```diff
@@ -12,10 +12,10 @@ uint8_t u8_to_dec(char *buf, uint8_t buflen, uint8_t value)
 	uint8_t num_digits = 0;
 	uint8_t digit;
 
-	while (buflen > 0 && divisor > 0) {
+	while ((buflen > 0) && (divisor > 0)) {
```

Comment:
>add enclosing parentheses enforcing and clarifying precedence of operators, improving code readability and maintainability, complying with advisory [misra-c2012-12.1] rule which states; The precedence of operators within expressions should be made explicit.

The precedence of the operators is clearly stated in the C standard, and these changes are not necessary. Over-parentheses is confusing and does not improve readability.